common/src/main/scala/azure/NotificationHubClient.scala

```diff
@@ -77,17 +77,29 @@ class NotificationHubClient(val notificationHubConnection: NotificationHubConnec
   }
 
   def sendNotification(rawPush: RawPush): Future[HubResult[Option[String]]] = {
-    val serviceBusTags = rawPush.tagQuery.map(tagQuery => "ServiceBusNotification-Tags" -> tagQuery).toList
-    logger.debug(s"Sending Raw Notification: $rawPush")
-    rawPush.post(
-      request(Endpoints.Messages)
-        .addHttpHeaders(rawPush.extraHeaders: _*)
-        .addHttpHeaders("ServiceBusNotification-Format" -> rawPush.format)
-        .addHttpHeaders(serviceBusTags: _*)
-    ).map {
-      case r if r.isSuccess => Right(r.header("Location"))
-      case r => Left(XmlParser.parseError(r))
+    def retry(count: Int): Future[HubResult[Option[String]]] = {
+      val serviceBusTags = rawPush.tagQuery.map(tagQuery => "ServiceBusNotification-Tags" -> tagQuery).toList
+      logger.debug(s"Sending Raw Notification: $rawPush")
+      rawPush.post(
+        request(Endpoints.Messages)
+          .addHttpHeaders(rawPush.extraHeaders: _*)
+          .addHttpHeaders("ServiceBusNotification-Format" -> rawPush.format)
+          .addHttpHeaders(serviceBusTags: _*)
+      ).flatMap {
+        case r if r.isSuccess =>
+          Future.successful(Right(r.header("Location")))
+        case r =>
+          val error = XmlParser.parseError(r)
+          if (count < 3 && error.reason.contains("Azure Notifications Hub Server Busy")) {
+            logger.error(s"Unable to push notification, try $count/3. Got response code: ${r.status} got the following error: $error for the notification $rawPush")
+            retry(count + 1)
+          } else {
+            logger.error(s"Unable to push notification, response code: ${r.status} got the following error: $error for the notification $rawPush")
+            Future.successful(Left(error))
+          }
+      }
     }
+    retry(1)
```

Comment:
Just to be clear, we retry twice? Maybe its a little clearer if this starts as 0 and you do `count < 2` on L92